### DISCUSSION TITLE
Problems getting Python interpreter up and running

### DISCUSSION THREAD

**Discussion by adrennhoff** [posit-dev/positron#4623] [Q&A]
I apologize in advance for being probably one of the least technically proficient users here. So, my apologies in advance for overlooking obvious issues. I initially experienced the issue related to needing ipykernel updated, but that seems to have been resolved. But I cannot get the Python interpreter to even start (the R interpreter works as expected).

I am going simple -- no virtual environments and, to my knowledge, only one Python install. The 'Select Interpreter' is only giving me one option:Python 3.9.12 (Conda: base)

![image](https://github.com/user-attachments/assets/2b4620a7-526e-48cf-b5a2-55a1f253ec9b)

In the console pane, however, code never executes. The code is grayed out (forever) and never leads to variables created. 

It's hard to describe the problem because, basically nothing happens. The commands are grayed out and never executed.

I have restarted and tried again several times. A few of the times, in the console window, I get this error:

No state found for session 'python-716f1102'

Here's the final weird piece. When I go to the command window and type "python -V" it tells me Python 3.9.19 which does not match the option I have when selecting the interpreter in Positron.

**Comment by juliasilge** [comment]
So sorry you are having trouble!

It does look like Positron thinks there are two Python interpreters available, both 3.9.12 and 3.9.19. Have you tried choosing the other one, that has the same version as what you see with `python -V` at the command line?

We may be able to find out more about what is going wrong if you are able to share logs. [Take a look at this info](https://github.com/posit-dev/positron/wiki/Troubleshooting#python-and-r-logs) and then check out your "Console" and "Python" logs, under "Output". The "Console" logs will look something like this:

![Screenshot 2024-09-10 at 1 48 23 PM](https://github.com/user-attachments/assets/02019578-ae11-4b89-a19e-7d9a0742ba99)

Under the `...` you can open this log file in an editor and then save it somewhere, to then attach as a file here on GitHub.

**Comment by adrennhoff** [comment]
Thanks so much for your response. First, in the Interpreter Select dropdown, I am unable to select the second Python. The "Start the Interpreter" button does not seem to do anything. 

Here is a paste of the log that I see...

2024-09-10 15:07:15.789 [info] [Positron] Connecting to Python 3.9.12 (Conda: base) kernel (pid 15356)
2024-09-10 15:07:15.789 [info] [Positron] Streaming log file: C:\Users\rennhoff\AppData\Local\Temp\kernel-9tvSVb\kernel.log
2024-09-10 15:07:15.790 [info] [Positron] Connecting to kernel sockets defined in C:\Users\rennhoff\AppData\Local\Temp\kernel-9tvSVb\connection.json...
2024-09-10 15:07:15.790 [info] [Positron] Control socket created (count = 1)
2024-09-10 15:07:15.790 [info] [Positron] Shell socket created (count = 2)
2024-09-10 15:07:15.790 [info] [Positron] Stdin socket created (count = 3)
2024-09-10 15:07:15.791 [info] [Positron] I/O socket created (count = 4)
2024-09-10 15:07:15.791 [info] [Positron] Heartbeat socket created (count = 5)
2024-09-10 15:07:15.791 [info] [Positron] Control socket connecting to tcp://127.0.0.1:48693...
2024-09-10 15:07:15.791 [info] [Positron] Shell socket connecting to tcp://127.0.0.1:50369...
2024-09-10 15:07:15.791 [info] [Positron] Stdin socket connecting to tcp://127.0.0.1:61044...
2024-09-10 15:07:15.791 [info] [Positron] I/O socket connecting to tcp://127.0.0.1:47151...
2024-09-10 15:07:15.792 [info] [Positron] Heartbeat socket connecting to tcp://127.0.0.1:54721...
[Positron] Error setting console width: Cannot invoke 'setConsoleWidth'; no UI comm is open. (undefined)
[Positron] Control socket still connecting (2s)
[Positron] Shell socket still connecting (2s)
[Positron] Stdin socket still connecting (2s)
[Positron] I/O socket still connecting (2s)
[Positron] Heartbeat socket still connecting (2s)
[Positron] Control socket still connecting (4s)
[Positron] Shell socket still connecting (4s)
[Positron] Stdin socket still connecting (4s)
[Positron] I/O socket still connecting (4s)
[Positron] Heartbeat socket still connecting (4s)
[Positron] Control socket connected to tcp://127.0.0.1:48693
[Python] *** Log started at Tue 09/10/2024 15:07:15.83 
[Python] *** Command line: 
[Python] C:\Users\rennhoff\Anaconda3\python.exe "c:\Program Files\Positron\resources\app\extensions\positron-python\python_files\positron\positron_language_server.py" -f C:\Users\rennhoff\AppData\Local\Temp\kernel-9tvSVb\connection.json --logfile C:\Users\rennhoff\AppData\Local\Temp\kernel-9tvSVb\kernel.log --loglevel error --session-mode console 
[Positron] Shell socket connected to tcp://127.0.0.1:50369
[Positron] Heartbeat socket connected to tcp://127.0.0.1:54721
[Positron] Stdin socket connected to tcp://127.0.0.1:61044
[Positron] I/O socket connected to tcp://127.0.0.1:47151
[Positron] Establishing socket listeners...
[Positron] Received initial heartbeat: heartbeat
[Positron] SEND kernel_info_request to Shell: {"buffers":[],"content":{},"header":{"msg_id":"a6f778fb-9b17-4554-b153-86b9add8a980","msg_type":"kernel_info_request","version":"5.0","date":"2024-09-10T20:07:21.180Z","session":"b5dfb8b1536618b6d64dcecb812ec3a7","username":"rennhoff"},"metadata":{},"parent_header":{}}
[Positron] Starting heartbeat check at 30 second intervals...
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND kernel_info_request: OK
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND heartbeat with timeout of 30 seconds
[Positron] SEND heartbeat with timeout of 30 seconds

**Reply by adrennhoff** [reply]
Just a quick follow-up. As best I can tell, there is only one Python installation on this PC and there is only one Python installation in the PATH.

**Comment by adrennhoff** [comment]
Well, I must really be doing something wrong somewhere. For fun, I tried to install and run Positron on my Dell laptop, as well (also running Windows 10). It is running Python 3.11.5. Again, I could not get the interpreter to properly execute any Python commands. Here is a copy/paste of the output file:

2024-09-10 21:15:25.701 [info] [Positron] Connecting to Python 3.11.5 (Conda: base) kernel (pid 21780)
2024-09-10 21:15:25.702 [info] [Positron] Streaming log file: C:\Users\rennhoff\AppData\Local\Temp\kernel-VfV5KT\kernel.log
2024-09-10 21:15:25.702 [info] [Positron] Connecting to kernel sockets defined in C:\Users\rennhoff\AppData\Local\Temp\kernel-VfV5KT\connection.json...
2024-09-10 21:15:25.703 [info] [Positron] Control socket created (count = 6)
2024-09-10 21:15:25.703 [info] [Positron] Shell socket created (count = 7)
2024-09-10 21:15:25.704 [info] [Positron] Stdin socket created (count = 8)
2024-09-10 21:15:25.704 [info] [Positron] I/O socket created (count = 9)
2024-09-10 21:15:25.705 [info] [Positron] Heartbeat socket created (count = 10)
2024-09-10 21:15:25.705 [info] [Positron] Control socket connecting to tcp://127.0.0.1:50680...
2024-09-10 21:15:25.705 [info] [Positron] Shell socket connecting to tcp://127.0.0.1:56010...
2024-09-10 21:15:25.706 [info] [Positron] Stdin socket connecting to tcp://127.0.0.1:55210...
2024-09-10 21:15:25.706 [info] [Positron] I/O socket connecting to tcp://127.0.0.1:62443...
2024-09-10 21:15:25.706 [info] [Positron] Heartbeat socket connecting to tcp://127.0.0.1:49133...
[Positron] Control socket still connecting (2s)
[Positron] Shell socket still connecting (2s)
[Positron] Stdin socket still connecting (2s)
[Positron] I/O socket still connecting (2s)
[Positron] Heartbeat socket still connecting (2s)
[Positron] Control socket still connecting (4s)
[Positron] Shell socket still connecting (4s)
[Positron] Stdin socket still connecting (4s)
[Positron] I/O socket still connecting (4s)
[Positron] Heartbeat socket still connecting (4s)
[Python] *** Log started at Tue 09/10/2024 21:15:25.92 
[Python] *** Command line: 
[Python] C:\Users\rennhoff\AppData\Local\anaconda3\python.exe "c:\Program Files\Positron\resources\app\extensions\positron-python\python_files\positron\positron_language_server.py" -f C:\Users\rennhoff\AppData\Local\Temp\kernel-VfV5KT\connection.json --logfile C:\Users\rennhoff\AppData\Local\Temp\kernel-VfV5KT\kernel.log --loglevel error --session-mode console 
[Positron] Control socket still connecting (6s)
[Positron] Shell socket still connecting (6s)
[Positron] Stdin socket still connecting (6s)
[Positron] I/O socket still connecting (6s)
[Positron] Heartbeat socket still connecting (6s)
[Positron] Control socket still connecting (8s)
[Positron] Shell socket still connecting (8s)
[Positron] Stdin socket still connecting (8s)
[Positron] I/O socket still connecting (8s)
[Positron] Heartbeat socket still connecting (8s)
[Positron] Control socket still connecting (10s)
[Positron] Shell socket still connecting (10s)
[Positron] Stdin socket still connecting (10s)
[Positron] I/O socket still connecting (10s)
[Positron] Heartbeat socket still connecting (10s)
[Positron] Control socket still connecting (12s)
[Positron] Shell socket still connecting (12s)
[Positron] Stdin socket still connecting (12s)
[Positron] I/O socket still connecting (12s)
[Positron] Heartbeat socket still connecting (12s)
[Positron] Control socket still connecting (14s)
[Positron] Shell socket still connecting (14s)
[Positron] Stdin socket still connecting (14s)
[Positron] I/O socket still connecting (14s)
[Positron] Heartbeat socket still connecting (14s)
[Positron] Control socket still connecting (16s)
[Positron] Shell socket still connecting (16s)
[Positron] Stdin socket still connecting (16s)
[Positron] I/O socket still connecting (16s)
[Positron] Heartbeat socket still connecting (16s)
[Positron] Control socket still connecting (18s)
[Positron] Shell socket still connecting (18s)
[Positron] Stdin socket still connecting (18s)
[Positron] I/O socket still connecting (18s)
[Positron] Heartbeat socket still connecting (18s)
[Positron] Control socket connect timed out after 20 seconds
[Positron] Shell socket connect timed out after 20 seconds
[Positron] Stdin socket connect timed out after 20 seconds
[Positron] I/O socket connect timed out after 20 seconds
[Positron] Heartbeat socket connect timed out after 20 seconds

**Comment by adrennhoff** [comment]
I'm just goin to go ahead and close the discussion. This morning I uninstalled Python and then reinstalled and everything worked as expected, so it must have been a quirk associated with my particular install.

**Reply by juliasilge** [reply]
Glad to hear it! Definitely let us know if you run into problems moving forward.
